### ISSUE TITLE
Health checker starts checking Pods before they are Ready, causing false negatives during startup

### ISSUE THREAD

**Issue by spencercjh** [kubeovn/endpoint-health-checker#2]
## Summary

The health checker begins checking Pods as soon as they reach `Running` phase with an IP address, without waiting for kubelet's readiness probe to pass. This causes false negatives during normal Pod startup, rolling updates, and scale-up operations.

## Current Behavior

**Code location**: `pkg/controller/podSet.go:28-41`

```go
func (ps *PodSet) AddOrUpdate(pod *corev1.Pod) {
    if pod.Status.Phase != corev1.PodRunning || pod.Status.PodIP == "" {
        return
    }

    if !shouldCheckPod(pod) {
        return
    }

    // Pod is immediately added to check queue
    // No verification that Pod is Ready
    ps.pods[pod.Status.PodIP] = &PodInfo{...}
}
```

**Problem:** The check `Phase == Running && PodIP != ""` is insufficient. A Pod can be `Running` with an IP but still starting up.

The correct Pod lifecycle should be:

```
Created → Pending → Running (has IP) → Ready (passed kubelet probe) → External health check
                                        ↑
                               Health checker should start here
```

Currently, health checking starts too early:

```
Created → Pending → Running (has IP) → External health check ✗ Too early!
                                     ↓
                              False negatives during startup
```

## Expected Behavior

Health checker should only check Pods that have **already passed kubelet's readiness probe** at least once:

```go
func (ps *PodSet) AddOrUpdate(pod *corev1.Pod) {
    // 1. Basic checks (existing)
    if pod.Status.Phase != corev1.PodRunning || pod.Status.PodIP == "" {
        return
    }

    if !shouldCheckPod(pod) {
        return
    }

    // 2. ⭐ New: Wait for Pod to be Ready
    if !isPodReady(pod) {
        klog.V(4).Infof("Skipping pod %s/%s: waiting for initial readiness probe to pass",
            pod.Namespace, pod.Name)
        return
    }

    // 3. Add to check queue
    ps.mu.Lock()
    defer ps.mu.Unlock()
    ps.pods[pod.Status.PodIP] = &PodInfo{...}
}

// Helper function to check Pod Ready condition
func isPodReady(pod *corev1.Pod) bool {
    for _, cond := range pod.Status.Conditions {
        if cond.Type == corev1.PodReady {
            return cond.Status == corev1.ConditionTrue
        }
    }
    return false
}
```

## Additional Notes

This is a common pattern in Kubernetes controllers. For example:
- **Horizontal Pod Autoscaler**: Only considers Ready Pods for scaling decisions
- **Deployment controller**: Only counts Ready Pods for rollout progress
- **Service endpoints**: Only includes Ready Pods

The health checker should follow the same pattern.

_If I've said anything incorrectly, please feel free to point it out. I haven't participated in the development of any Kubernetes open source projects._
